litestar/contrib/repository/__init__.py

```diff
@@ -1,14 +1,20 @@
-from __future__ import annotations
-
-from .abc import AbstractAsyncRepository, AbstractSyncRepository
-from .exceptions import ConflictError, NotFoundError, RepositoryError
-from .filters import FilterTypes
-
-__all__ = (
-    "AbstractAsyncRepository",
-    "AbstractSyncRepository",
-    "ConflictError",
-    "FilterTypes",
-    "NotFoundError",
-    "RepositoryError",
-)
+from litestar.utils import warn_deprecation
+
+
+def __getattr__(attr_name: str) -> object:
+    from litestar import repository
+
+    for k in repository.__all__:
```

Comment:
Why warn for all members instead of just the one that's actually being accessed? Getting a warning about a deprecated import if you didn't actually try to import that thing sounds confusing. 
Applies to all the other files as well.

Reply:
fixed